conf/templates/README.md

```diff
@@ -0,0 +1,188 @@
+[ec2]
+...
+cluster_template = example
+...
+```
+The configured value must match the name of a subdirectory under
+`conf/templates`
+
+```
+~$ ls -1a fluo-muchos/conf/templates/example
+.
+..
+.devices
+metrics.json
+namenode.json
+resourcemanager.json
+zookeeper.json
+worker.json
+```
+The subdirectory will contain one or more user-defined EC2 launch
+templates (*\*.json*) for your various host types, and it will
+include a *.devices* file specifying the desired mount points for all
+data volumes (excluding root volumes, as they are mounted
+automatically)
+
+### Defining EC2 launch templates and device mounts for your hosts
+
+#### Launch Templates: *{service-name}.json*
+
+Each JSON file represents a standard EC2 launch request, and each file
+name must match one of the predefined muchos service names, as
+defined in the **nodes** section of *muchos.props*. E.g.,
+```
+... 
+[nodes]
+leader1 = namenode,resourcemanager,accumulomaster
+leader2 = metrics,zookeeper
+worker1 = worker
+worker2 = worker
+worker3 = worker
+worker4 = worker
+```
+In template mode, the first service listed for a given host denotes the
+template to be selected for its launch.
+
+Based on the example given above:
+* **leader1** selects **namenode.json**
+* **leader2** selects **metrics.json**
+* **worker1** selects **worker.json**
+* and so on...
+  
+For example, *namenode.json* might be defined as follows...
+```
+{
+    "KeyName": "${key_name}",
+    "BlockDeviceMappings": [
+        {
+            "DeviceName": "/dev/sda1",        # Here, /dev/sda1 denotes the root volume
+            "Ebs": {
+                "DeleteOnTermination": true,
+                "VolumeSize": 40,
+                "VolumeType": "gp2"
+            }
+        },
+        {
+            "DeviceName": "/dev/sdf",          # Here, /dev/sdf (a.k.a "/dev/xvdf") denotes
+            "Ebs": {                           # our single EBS-backed data volume
+                "DeleteOnTermination": true, 
+                "VolumeSize": 500,
+                "VolumeType": "gp2"
+            } 
+        }
+    ], 
+    "ImageId": "${aws_ami}",
+    "InstanceType": "m4.4xlarge",
+    "NetworkInterfaces": [
+        {
+            "DeviceIndex": 0,
+            "AssociatePublicIpAddress": ${associate_public_ip},
+            "Groups": [
+                "${security_group_id}"
+            ]
+        }
+    ],
+    "EbsOptimized": true,
+    "InstanceInitiatedShutdownBehavior": "${shutdown_behavior}"
+}
+```
+*Property Placeholders*
+
+The `${property name}` placeholders demonstrated above are optional and
+are intended to simplify template creation and reduce maintenance burden.
+They allow any matching properties from the **ec2** section of *muchos.props*
+to be interpolated automatically.
+
+If needed, you may also define your own custom properties and have them be injected
+automatically by simply adding them to the **ec2** section of
+*muchos.props* and to your templates
+
+#### Device Mounts: *.devices*
```

Comment:
It's not a good idea to use the dotfile naming convention here for this file. It will create files which are hidden in \*nix systems.